### ISSUE TITLE
Files deleted within Atom cannot be 'Put Back' in Finder

### ISSUE THREAD

**Issue by Sintendo** [atom/electron#2001]
Delete a file within Atom. Go to Trash using Finder. Right click the file. There will be no 'Put Back' option allowing you to restore it, and the option will be disabled in the menu bar.

Relevant code: https://github.com/atom/electron/blob/c97c3fb9a1f28d56ba2b6cc7090ab438204ec78e/atom/common/platform_util_mac.mm#L142

Unfortunately, it seems there is currently no native call in OSX which allows you to delete a file while allowing it to be restored later. The only way around this is by executing AppleScript to tell Finder to delete the file for you... Yuck!

**Comment by Sintendo** [comment]
Does that really work? This was one of the functions I've tried, and I found that files deleted with it still could not be restored.

**Comment by deepak1556** [comment]
@Sintendo yup works just tested on OSX(10.9.5) .

**Comment by Sintendo** [comment]
Very odd. I'm running 10.10.3.

**Comment by Sintendo** [comment]
Running Atom v1.5.3 on OS X 10.11.3, but I still don't get the 'Put Back' options for files deleted in Atom.

**Comment by bertofer** [comment]
Same here, still doesn't work for me.

**Comment by danielbayley** [comment]
I'm getting the same behaviour…

macOS 10.12.1
Atom 1.13.0-beta2 x64
Electron 1.3.6

**Comment by rtgreen86** [comment]
Same problem. There is no 'put back' button.

macOS 10.13.16
Atom 1.32.2
and Electron 3.0.8

**Comment by sindresorhus** [comment]
"Put Back" not working is a macOS bug and not something Electron can fix.

See: https://github.com/sindresorhus/macos-trash/issues/4

**Comment by 3otes** [comment]
Thanks for nothing! No warning, no restore! Just an innocent looking option to "discard all changes" with the option to "Undo last discard" that does not work!!! Yesterday I last more than a years work because I was trying to commit the last weeks work to a Git Lab repository instead of Git Hub. Atom inserted all my work that was not committed in Git Lab in the Git sidebar even though I only have one project folder open. So I tried the "discard all changes" thinking that would clear out all the items I did not want to commit, hey what could possible go wrong, when there is a "Undo last discard" option.... So how do I explain to my employer that the last weeks work is gone? Or that the editor I am using can permanently DELETE a years work with no restore option?
